lib/samples/animate_3d_graphic/animate_3d_graphic.dart

```diff
@@ -0,0 +1,647 @@
+
+  // The graphic for the plane.
+  Graphic? _planeGraphic;
+
+  // The camera controller that follows the plane.
+  late OrbitGeoElementCameraController _cameraController;
+
+  // Whether the animation is currently playing.
+  var _isPlaying = false;
+
+  // The current frame index in the animation.
+  var _currentFrameIndex = 0;
+
+  // The list of animation frames.
+  var _frames = <Frame>[];
+
+  // The ticker that drives the animation.
+  late final Ticker _ticker;
+
+  // The currently selected mission.
+  var _currentMission = Mission.grandCanyon;
+
+  // The selected animation speed.
+  var _animationSpeed = AnimationSpeed.medium;
+
+  // Camera controller settings.
+  double _cameraDistance = 1000;
+  double _cameraHeading = 0;
+  double _cameraPitch = 0;
+  var _autoHeading = true;
+  var _autoPitch = false;
+  var _autoRoll = false;
+
+  // State variables for real-time telemetry tracking.
+  double _altitude = 0;
+  double _heading = 0;
+  double _pitch = 0;
+  double _roll = 0;
+
+  // The current progress of the animation.
+  double get _progress {
+    if (_frames.isEmpty) return 0;
+    return _currentFrameIndex / _frames.length;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    // Start the animation ticker.
+    _ticker = createTicker(_onTick)..start();
+  }
+
+  @override
+  void dispose() {
+    // Cleanup the ticker.
+    _ticker.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a scene view to the widget tree and set a controller.
+                  child: ArcGISSceneView(
+                    controllerProvider: () => _sceneViewController,
+                    onSceneViewReady: onSceneViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    // Button to open mission settings.
+                    ElevatedButton(
+                      onPressed: _showMissionSettings,
+                      child: const Text('Mission'),
+                    ),
+                    // Play/pause button.
+                    IconButton(
+                      onPressed: _toggleAnimation,
+                      icon: Icon(_isPlaying ? Icons.pause : Icons.play_arrow),
+                    ),
+                    // Button to open camera settings.
+                    ElevatedButton(
+                      onPressed: _showCameraSettings,
+                      child: const Text('Camera'),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Show real-time telemetry data.
+            if (_planeGraphic != null)
```

Comment:
I think it's a little weird to use `_planeGraphic` here. One problem is that when `_planeGraphic` is set down in onSceneViewReady, it does not use `setState()`. This only works as intended because onSceneViewReady happens to use `setState()` to set `_ready` -- if that wasn't also happening, Flutter wouldn't know to call a rebuild.

So actually a real simple improvement here would be to use `if (_ready)` rather than `if (_planeGraphic != null)`. That ensures that the plane is loaded, as well as that the frames are loaded and running.

Reply:
Well actually I would use this widget: `Visibility(visible: _ready, child: SafeArea(...))`